### PR TITLE
mistake in the requierments fixed

### PR DESCRIPTION
**Pull request by 1kamma**

### CODE REVIEW

File: promotion.py [hunk 1]
```diff
@@ -44,7 +45,7 @@ class Promotion:
 
     def __init__(self, content: str, start_date: datetime, end_date: datetime, update_date: datetime, items: List[Item],
                  promo_func: callable, club_id: ClubID, promotion_id: float, max_qty: int,
-                 allow_multiple_discounts: bool, reward_type: RewardType):
+                 allow_multiple_discounts: bool, reward_type: RewardType, type_file: str = "excel"):
```
Comment (korenLazar) [inline promotion.py:48]: Why do we need it as a member?

File: promotion.py [hunk 2]
```diff
@@ -75,28 +77,43 @@ def write_promotions_to_csv(promotions: List[Promotion], output_filename: str) -
...
+            for promo in promotions:
+                prms = dict_promos([get_promotion_row_in_csv(promo, item) for item in promo.items], columns)
+                if prms:
+                    dt = dt.append(prms, True)
+                else:
+                    continue
+            dt.to_excel(xl, index=False, sheet_name="name")
+
+
+def dict_promos(promos: list, columns: list):
```
Comment (korenLazar) [inline promotion.py:115]: Documentation is needed.

File: promotion.py [hunk 3]
```diff
@@ -245,7 +263,8 @@ def main_latest_promos(store_id: int, load_xml: bool, chain: SupermarketChain, l
     promotions: List[Promotion] = get_available_promos(chain, store_id, load_xml, load_promos)
     promotions.sort(key=lambda promo: (max(promo.update_date.date(), promo.start_date.date()), promo.start_date -
                                        promo.end_date), reverse=True)
-    write_promotions_to_csv(promotions, f'results/{repr(type(chain))}_promos_{store_id}.csv')
+    ex_file = f'results/{repr(type(chain))}_promos_{store_id}{file_type}'
```
Comment (korenLazar) [inline promotion.py:266]: Here the dot should be placed hard coded

File: main.py [hunk 4]
```diff
@@ -78,12 +78,19 @@
                         help='The name of the requested chain',
                         choices=chain_dict.keys(),
                         )
+    parser.add_argument('--type',
```
Comment (korenLazar) [inline main.py:81]: Rename to be more meaningful, e.g. '--promotions-file-type'.
```suggestion
    parser.add_argument('--promotions-file-type',
```

File: main.py [hunk 5]
```diff
@@ -78,12 +78,19 @@
...
+                        choices=("excel", "csv"),
+                        default='excel',
+                        
+        help="a switch flag to set set the output file to a CSV file",
```
Comment (korenLazar) [inline main.py:85]: We can make it more informative, e.g.:
'The extension/type of the promotions output file'.
You wrote 'set' twice.

**Review by korenLazar — COMMENTED** [review]
Overall, a very elegant solution.
I wrote some minor comments.

File: promotion.py (hunk 1 above)
Comment (1kamma) [reply promotion.py:48]: to pass it farword, for the last file

**Comment by 1kamma** [comment]
all have been corrected

**Comment by korenLazar** [comment]
Rethinking about it, I think it would be much clearer if we stick with simply file extension (and not file extension/type), as it can lead to confusions.
The documentation of the "dict_promos" function is still not clear. What do you mean by "the place in the xml file"? what is this p (inside the for loop) exactly?
Also, please change the name of the function to something more meaningful.

**Comment by 1kamma** [comment]
fixed some mistakes, most of them are typos

**Pull request merged**